libyul/AsmParser.cpp

```diff
@@ -567,18 +567,20 @@ std::variant<Literal, Identifier> Parser::parseLiteralOrIdentifier(bool _unlimit
 			break;
 		}
 
+		auto const beginOfLiteralLoc = currentLocation();
 		Literal literal{
 			createDebugData(),
 			kind,
-			valueOfLiteral(currentLiteral(), kind, _unlimitedLiteralArgument && kind == LiteralKind::String),
-			kind == LiteralKind::Boolean ? m_dialect.boolType : m_dialect.defaultType
+			valueOfLiteral(currentLiteral(), kind, _unlimitedLiteralArgument && kind == LiteralKind::String)
 		};
 		advance();
 		if (currentToken() == Token::Colon)
 		{
 			expectToken(Token::Colon);
 			updateLocationEndFrom(literal.debugData, currentLocation());
-			literal.type = expectAsmIdentifier();
+			auto const typedLiteralLocation = SourceLocation::smallestCovering(beginOfLiteralLoc, currentLocation());
```

Comment:
This was missed from https://github.com/ethereum/solidity/pull/15332#discussion_r1731541551. As far as I can see it's the whole literal, not just the beginning of it so the current name is misleading.

```suggestion
			auto const typedLiteralLocation = SourceLocation::smallestCovering(literalLocation, currentLocation());
```